internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -2288,3 +2301,43 @@ func (r *AgentServiceConfigReconciler) getImageService(ctx context.Context, log
 	}
 	return imageServiceURL
 }
+
+func (r *AgentServiceConfigReconciler) deployAgentInstallCRDs(secret *corev1.Secret, namespace string, log logrus.FieldLogger) error {
+	// Get in-cluster client
+	kubeClient, err := r.K8sApiExtensionsClientFactory.CreateFromInClusterConfig()
+	if err != nil {
+		return err
+	}
+
+	// Fetch all agent-install CRDs in cluster
+	crdClient := kubeClient.ApiextensionsV1().CustomResourceDefinitions()
+	crds, err := crdClient.List(context.TODO(), metav1.ListOptions{
```

Comment:
Why not pass the context from the caller in and use that?

Reply:
Had an issue with the context for ApiextensionsV1 with creating CRDs, I'll try to reproduce to see the specific error.